app/src/main/java/net/vonforst/evmap/storage/ChargeLocationsDao.kt

```diff
@@ -218,6 +229,19 @@ class ChargeLocationsRepository(
         return CacheLiveData(dbResult, apiResult)
     }
 
+    /**
+     * Numeric date for database limit required limit on some APIs
+     */
+    fun afterDate(dataSource: String): Long {
+        val after = Calendar.getInstance()
+        var days = 1
+        if (prefs.dataSource.equals("openchargemap")) {
+            days = 300
+        }
+        after.add(Calendar.DATE, -days)
+        return after.toInstant().toEpochMilli()
+    }
+
```

Comment:
As I said, this duration (1 or 300 days) should be something that the API class (implementation of the `ChargepointApi` interface) itself returns, rather than making it another `if` clause that needs to be updated in case we add another API.

Also the calculation could probably be implemented in a more concise way with the new `java.time` APIs that we already use in other places, e.g. the API would return a duration `Duration.ofDays(1)`, and then the `ChargeLocationsRepository` would calculate a value `Instant.now().plus(duration).toEpochMilli()`.

Reply:
I tried Java-ish solution first and then found this which should be equivalent - also, I now moved the keep-days limit to their own classes. I don't have the European data key so I can't test that all though.

Reply:
Still, `Instant.now().minus(duration).toEpochMilli()` is shorter than creating a `Calendar` instance, subtracting days from that and then converting it to milliseconds (and you're actually converting the `Calendar` to an `Instant` in that process with the `toInstant()` call, so you could have directly used and `Instant` in the first place).

Also, what the API returns could actually be a `Duration` (then it could even be less than a day if necessary... especially useful for testing).

Reply:
Ok how is this now?

Reply:
The first part is fixed, I will do the second now